src/main/java/core/basesyntax/service/impl/TransactionParserImpl.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.Transaction;
+import core.basesyntax.service.TransactionParser;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TransactionParserImpl implements TransactionParser {
+    private static final String SEPARATOR = ",";
+    private static final int EXPECTED_PARTS = 3;
+    private static final int OPERATION_TYPE = 0;
+    private static final int FRUIT_NAME = 1;
+    private static final int QUANTITY = 2;
+
+    @Override
+    public List<Transaction> parse(List<String> rawData) {
+        if (rawData == null || rawData.isEmpty()) {
+            throw new IllegalArgumentException("Invalid transaction line: " + rawData);
+        }
+        List<Transaction> transactions = new ArrayList<>();
+        for (String line: rawData) {
+            String[] parts = line.split(SEPARATOR);
+            if (parts.length != EXPECTED_PARTS) {
+                throw new IllegalArgumentException("Invalid transaction format: " + line);
```

Comment:
The error message for an invalid transaction format should provide more context about the expected format, e.g., 'Invalid transaction format. Each line should contain exactly 3 parts separated by commas: operation, fruit name, and quantity.'